src/mjolnir/pbfgraphparser.cc

```diff
@@ -2291,16 +2290,17 @@ struct graph_parser {
     osmdata_.edge_count -= intersection; // more accurate but undercounts by skipping lone edges
   }
 
-  void way(const osmium::Way& way) {
-    changeset(way.changeset());
-
-    osmid_ = static_cast<uint64_t>(way.id());
-    // unsorted extracts are just plain nasty, so they can bugger off!
-    if (osmid_ < last_way_) {
-      throw std::runtime_error("Detected unsorted input data");
-    }
-    last_way_ = osmid_;
+  struct Way {
+    uint64_t osmid;
+    Tags tags;
+    std::vector<uint64_t> nodes;
+    uint64_t changeset_id;
+  };
```

Comment:
Probably not the best place to define a structure, but it is nice to have it declared right near to where it is used. Please let me know if this declaration should be moved somewhere else in this 5k lines file ;)